dcos_tunnel/cli.py

```diff
@@ -611,11 +621,23 @@ def _vpn(port, config_file, user, privileged, ssh_port, host,
         clientconfigfile, clientconfigpath = config_tup
         clientfile, clientpath = client_tup
 
+        # Test if docker can be run on remote node
+        # if not try adding sudo and retry
+        scom = '{} version'.format(docker_cmd)
+        _, query_stdout, _ = client.exec_command(scom, get_pty=True)
+        if query_stdout.channel.recv_exit_status() != 0\
+                and not docker_cmd.startswith('sudo'):
+            docker_cmd = 'sudo ' + docker_cmd
+            scom = '{} version'.format(docker_cmd)
+            _, query_stdout, _ = client.exec_command(scom, get_pty=True)
```

Comment:
Since it seems like this this `client.exec_command()` and subsequent check for `recv_exit_status()` and raising exception is done everywhere. Maybe pull this out into a separate function?